commit b4710078310e9a5243932ab90b77b760fa97faa3
Author: Guillaume Bougard <gbougard@teclib.com>
Date:   Tue Apr 30 12:56:06 2019 +0200

    fix: HTTP server plugins base configuration folder

diff --git a/lib/FusionInventory/Agent/HTTP/Server/Plugin.pm b/lib/FusionInventory/Agent/HTTP/Server/Plugin.pm
index 5d570aab5..1076eb30e 100644
--- a/lib/FusionInventory/Agent/HTTP/Server/Plugin.pm
+++ b/lib/FusionInventory/Agent/HTTP/Server/Plugin.pm
@@ -23,6 +23,10 @@ sub new {
 
     bless $self, $class;
 
+    # Import _confdir from agent configuration
+    $self->{_confdir} = $self->{server}->{agent}->{config}->{_confdir}
+        if $self->{server};
+
     # Check _confdir imported from FusionInventory::Agent::Config
     unless ($self->{_confdir} && -d $self->{_confdir}) {
         # Set absolute confdir from default if replaced by Makefile otherwise search
